lib/content-services/src/lib/tree/components/tree.component.spec.ts

```diff
@@ -0,0 +1,259 @@
+        expect(collapseSpy).toHaveBeenCalledWith(component.treeService.treeNodes[0], treeNodesMockExpanded);
+    });
+
+    it('should load more subnodes and remove load more button when load more button is clicked', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        spyOn(component.treeService, 'getSubNodes').and.returnValue(of({pagination: {}, entries: Array.from(singleNode)}));
+        const loadMoreBtn = fixture.debugElement.query(By.css('.adf-tree-load-more-button adf-icon')).nativeElement;
+        const appendSpy = spyOn(component.treeService, 'appendNodes').and.callThrough();
+        loadMoreBtn.dispatchEvent(new Event('click'));
+        fixture.whenStable();
+        fixture.detectChanges();
+        const loadMoreNodes = component.treeService.treeNodes.find((node: TreeNode) => node.nodeType === TreeNodeType.LoadMoreNode);
+        expect(appendSpy).toHaveBeenCalledWith(component.treeService.treeNodes[0], Array.from(singleNode));
+        expect(loadMoreNodes).toBeUndefined();
+    });
+
+    it('selection should be disabled by default, no checkboxes should be displayed', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        const nodeCheckboxes = fixture.debugElement.queryAll(By.css('mat-checkbox'));
+        expect(nodeCheckboxes.length).toEqual(0);
+        expect(component.selectableNodes).toEqual(false);
+    });
+
+    describe('Tree nodes selection tests', () => {
+        beforeEach(() => {
+            component.selectableNodes = true;
+        });
+
+        it('should display checkboxes when selection is enabled', () => {
+            component.refreshTree();
+            fixture.detectChanges();
+            const nodeCheckboxes = fixture.debugElement.queryAll(By.css('mat-checkbox'));
+            const selectableNodes = component.treeService.treeNodes.filter((node: TreeNode) => node.nodeType !== TreeNodeType.LoadMoreNode);
+            expect(nodeCheckboxes.length).toEqual(selectableNodes.length);
+        });
+
+        it('should update selection when leaf node is selected', () => {
+            spyOn(component.treeService, 'getSubNodes').and.returnValue(of({ pagination: {}, entries: Array.from(treeNodesMock) }));
+            fixture.detectChanges();
+            const nodeCheckboxes = fixture.debugElement.queryAll(By.css('mat-checkbox'));
+            nodeCheckboxes[0].nativeElement.dispatchEvent(new Event('change'));
+            expect(component.treeNodesSelection.isSelected(component.treeService.treeNodes[0])).toBeTrue();
+        });
+
+        it('should update selection of each child node when parent node is selected and deselected', () => {
+            spyOn(component.treeService, 'getSubNodes').and.returnValue(of({ pagination: {}, entries: Array.from(treeNodesChildrenMockExpanded) }));
+            fixture.detectChanges();
+            const nodeCheckboxes = fixture.debugElement.queryAll(By.css('mat-checkbox'));
```

Comment:
these two lines can be turned into the helper function like "tickCheckbox(index: number)" or something similar, so that tests are more readable

Reply:
Added